File: src/main/java/org/ffpy/plugin/coding/util/StreamUtils.java
package org.ffpy.plugin.coding.util;

import java.util.HashSet;
import java.util.Set;
import java.util.function.Function;
import java.util.function.Predicate;
import java.util.stream.Stream;

/**
 * {@link Stream}工具类
 */
public class StreamUtils {

    public static <T, R> Predicate<T> distinct(Function<T, R> valueExtract) {
        Set<R> set = new HashSet<>();
        return value -> set.add(valueExtract.apply(value));
    }
}

File: src/main/kotlin/org/ffpy/plugin/coding/util/StreamUtils.kt
package org.ffpy.plugin.coding.util

import java.util.*
import java.util.function.Function
import java.util.function.Predicate

/**
 * [Stream]工具类
 */
object StreamUtils {
    fun <T, R> distinct(valueExtract: Function<T, R>): Predicate<T> {
        val set: MutableSet<R> = HashSet()
        return Predicate { value: T -> set.add(valueExtract.apply(value)) }
    }
}